docs/proposals/20240807-in-place-updates-implementation-notes.md

```diff
@@ -86,3 +86,173 @@ sequenceDiagram
     MS2 (NewMS)-->>MS Controller: Yes, M1!
     MS Controller->>M1: Remove annotation ".../pending-acknowledge-move": ""
 ```
+
+## Notes about in-place update implementation for KubeadmControlPlane
+
+- In-place updates respect the existing control plane update strategy:
+  - KCP controller uses `rollingUpdate` strategy with `maxSurge` (0 or 1)
+  - When `maxSurge` is 0, no new machines are created during rollout - only in-place updates or scale down
+  - When `maxSurge` is 1:
+    - Controller first scales up by creating one new machine to maximize fault tolerance
+    - Once at `maxReplicas` (desiredReplicas + 1), evaluates whether to in-place update or scale down old machines
+    - For each old machine needing rollout: if eligible for in-place update, performs in-place; otherwise scales down
+    - This pattern repeats until all machines are up-to-date, then scales back to desired replica count
+  
+- The implementation respects the existing set of responsibilities:
+  - KCP controller manages control plane Machines directly
+    - KCP controller enforces `maxSurge` limits during rolling updates
+    - KCP controller decides when to scale up, scale down, or perform in-place updates
+    - KCP controller runs preflight checks to ensure control plane is stable before in-place updates
+    - KCP controller calls `CanUpdateMachine` hook to verify if extensions can handle the changes
+    - When in-place update is possible, KCP controller triggers the update by writing desired state
```

Comment:
```suggestion
    - When in-place update is possible, the KCP controller triggers the update by writing the desired state
```

isn't it reconciling instead of writing?